From 7b210f66eee10ea647ea6fcad67288794dc7cdb8 Mon Sep 17 00:00:00 2001
From: Todd Poynor <toddpoynor@google.com>
Date: Fri, 13 Jul 2012 15:37:53 -0700
Subject: [PATCH 594/944] power: smb347: enable wake for interrupt

Based on code from HongMin Son <hongmin.son@samsung.com>

Change-Id: I6669f9bfdf40e0de396b9f42732efc101d9343dd
Signed-off-by: Todd Poynor <toddpoynor@google.com>
Signed-off-by: HongMin Son <hongmin.son@samsung.com>
---
 drivers/power/smb347-charger.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/power/smb347-charger.c b/drivers/power/smb347-charger.c
index f429609..f03dae3 100644
--- a/drivers/power/smb347-charger.c
+++ b/drivers/power/smb347-charger.c
@@ -915,6 +915,10 @@ static int smb347_irq_init(struct smb347_charger *smb)
 	if (ret < 0)
 		goto fail_gpio;
 
+	ret = enable_irq_wake(irq);
+	if (ret)
+		pr_err("%s: failed to enable wake on irq %d\n", __func__, irq);
+
 	ret = smb347_set_writable(smb, true);
 	if (ret < 0)
 		goto fail_irq;
@@ -945,6 +949,7 @@ static int smb347_irq_init(struct smb347_charger *smb)
 fail_readonly:
 	smb347_set_writable(smb, false);
 fail_irq:
+	disable_irq_wake(irq);
 	free_irq(irq, smb);
 fail_gpio:
 	gpio_free(pdata->irq_gpio);
@@ -1391,6 +1396,7 @@ static int smb347_remove(struct i2c_client *client)
 
 	if (client->irq) {
 		smb347_irq_disable(smb);
+		disable_irq_wake(client->irq);
 		free_irq(client->irq, smb);
 		gpio_free(smb->pdata->irq_gpio);
 	}
-- 
1.8.0

